File: src/cpp/Carnivore.cpp
/*
 * File:   Carnivore.cpp
 * Author: David KALONJI
 * Email : kalonjidvd@gmail.com
 *
 * Created on 26 aug 2016, 13:49
 * Last modified on Apr 20 16:52 2017
 */

/**
 Implementation du type concret Carnivore de regimeAlimentaire.
 */

#include "Carnivore.h"

Carnivore::Carnivore(){} // Constructeur

void Carnivore::seNourrir(Faune* consommateur, ObjetJungle* consommable)
{
    Faune* laProie = dynamic_cast<Faune*>(consommable);

    if(consommateur->getEnergie() != 0)
    {
        consommateur->augmenterDEnergie(laProie->getEnergie());
        laProie->setEnergie(0);
    }
}

Carnivore::~Carnivore(){} // Destructeur

File: src/java/Carnivore.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * File:   Carnivore.java
 * Author: David KALONJI
 * Email : kalonjidvd@gmail.com
 *
 * Created on 27 août 2016, 13:49
 * Last modified on Apr 21 16:40 2017
 */

/**
 * Classe concrete Carinivore implementant l'interface RegimeAlimentaire.
 */
public class Carnivore implements RegimeAlimentaire{

    /**
     * Implementation de la methode abstraite de l'interface RegimeAlimentaire.
     *
     * @param consommateur
     * @param consommable
     */
    public void seNourrir(Faune consommateur, ObjetJungle consommable){
        if(consommateur.getEnergie() != 0)
        {
            consommateur.augmenterDEnergie(((Faune)consommable).getEnergie());
            ((Faune)consommable).setEnergie(0);
        }
    }
}
